core/unit_test/TestMathematicalFunctions.hpp

```diff
@@ -1953,8 +1956,7 @@ TEST(TEST_CATEGORY, mathematical_functions_nextafter_fp16) {
 #if defined(KOKKOS_ENABLE_CUDA) && defined(KOKKOS_COMPILER_MSVC)
   GTEST_SKIP() << "FIXME MSVC nextafter for half precision "
                   "not implemented yet";
-#endif
-
+#else
```

Comment:
why bother with the `#else` (why not just `#endif` and unguarded code below)

Reply:
Because for MSVC with CUDA, the following macros are true. However, we do not have fp16 nextafter and cannot call fp16 test function.

```C++
#if defined(KOKKOS_HALF_T_IS_FLOAT) && !KOKKOS_HALF_T_IS_FLOAT
#if defined(KOKKOS_BHALF_T_IS_FLOAT) && !KOKKOS_BHALF_T_IS_FLOAT
```